plugins/httpapi/zabbix.py

```diff
@@ -171,6 +175,47 @@ def send_request(self, data=None, request_method="POST", path="/api_jsonrpc.php"
                 raise ConnectionError("Invalid JSON response: %s" % value)
 
             if "error" in json_data:
+                if "re-login" in json_data["error"]["data"]:
+                    # Get this response from Zabbix when we switch username to execute REST API
+                    if not self.auth_key:
+                        # Provide "fake" auth so netcommon.connection does not replace our headers
+                        self.connection._auth = {'auth': 'fake'}
+                    # Need to login with new username/password
+                    self.login(self.username, self.password)
```

Comment:
Maybe instead of caching the password yourself in the httpapi, you could always get the latest versions from the connection objects options:

```python
self.login(self.connection.get_option('remote_user'), self.connection.get_option('password'))
```
Making the above change to your code seems to make it work for my use-case

Reply:
I don't mind. Let me test and make the change.

Reply:
Tested, this works for my use-case!

Reply:
Thanks for testing, merging.